src/error.rs

```diff
@@ -10,9 +12,10 @@ pub enum Error<E> {
     /// Permanent means that it's impossible to execute the operation
     /// successfully. This error is immediately returned from `retry()`.
     Permanent(E),
-    /// Transient means that the error is temporary, the operation should
-    /// be retried according to the backoff policy.
-    Transient(E),
+    /// Transient means that the error is temporary. If the second argument is `None`
+    /// the operation should be retried according to the backoff policy, else after
+    /// the specified duration. Useful for handling ratelimits like a HTTP 429 response.
+    Transient((E, Option<Duration>)),
```

Comment:
You don't need the double parentheses, enum variants can have multiple "arguments" just by themselves.